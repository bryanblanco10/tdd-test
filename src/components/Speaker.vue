<template>
  <div>Speaker</div>
</template>

<script>
export default {
  props: {
    speaker: {
      type: Object,
      default: () => {},
    },
  },
};
</script>

<style></style>
