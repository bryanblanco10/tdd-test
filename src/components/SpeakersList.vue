<template>
  <div>
    <speaker
      v-for="(speaker, index) in speakers"
      :key="index"
      :speaker="speaker"
    />
  </div>
</template>

<script>
import Speaker from "./Speaker";

export default {
  components: {
    Speaker,
  },
  props: {
    speakers: {
      type: Array,
      default: () => [],
    },
  },
};
</script>

<style></style>
