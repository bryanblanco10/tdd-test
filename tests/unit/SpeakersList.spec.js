import { shallowMount } from "@vue/test-utils";
import SpeakersList from "@/components/SpeakersList";
import Speaker from "@/components/Speaker";

describe("SpeakersList", () => {
  let props;
  let speaker = {
    username: "@bryan",
    name: "Bryan",
  };

  beforeEach(() => {
    props = {
      speakers: [speaker],
    };
  });

  const build = () => {
    const wrapper = shallowMount(SpeakersList, {
      propsData: props,
    });
    return {
      wrapper,
      Speakers: () => wrapper.findAllComponents(Speaker),
    };
  };

  it("should render the component", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render the correct", async () => {
    const { wrapper, Speakers } = build();
    expect(Speakers().length).toBe(1);
    await wrapper.setProps({
      speakers: [
        speaker,
        {
          ...speaker,
          username: "@username",
        },
      ],
    });
    expect(Speakers().length).toBe(2);
  });

  it("should send speaker data to the childs", () => {
    const { Speakers } = build();
    expect(Speakers().at(0).vm.speaker).toBe(speaker);
  });
});
