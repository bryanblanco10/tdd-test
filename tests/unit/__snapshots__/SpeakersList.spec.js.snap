// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakersList should render the component 1`] = `
<div>
  <speaker-stub speaker="[object Object]"></speaker-stub>
</div>
`;
